hudi-flink/src/main/java/org/apache/hudi/sink/bootstrap/BootstrapFunction.java

```diff
@@ -168,33 +178,74 @@ private HoodieFlinkTable getTable() {
     long start = System.currentTimeMillis();
+
     BaseFileUtils fileUtils = BaseFileUtils.getInstance(this.hoodieTable.getBaseFileFormat());
-    List<HoodieBaseFile> latestBaseFiles =
-        HoodieIndexUtils.getLatestBaseFilesForPartition(partitionPath, this.hoodieTable);
-    LOG.info("All baseFile in partition {} size = {}", partitionPath, latestBaseFiles.size());
+    Schema schema = new TableSchemaResolver(this.hoodieTable.getMetaClient()).getTableAvroSchema();
 
     final int parallelism = getRuntimeContext().getNumberOfParallelSubtasks();
     final int maxParallelism = getRuntimeContext().getMaxNumberOfParallelSubtasks();
     final int taskID = getRuntimeContext().getIndexOfThisSubtask();
-    for (HoodieBaseFile baseFile : latestBaseFiles) {
-      boolean shouldLoad = KeyGroupRangeAssignment.assignKeyToParallelOperator(
-          baseFile.getFileId(), maxParallelism, parallelism) == taskID;
-
-      if (shouldLoad) {
-        LOG.info("Load records from file {}.", baseFile);
-        final List<HoodieKey> hoodieKeys;
-        try {
-          hoodieKeys =
-              fileUtils.fetchRecordKeyPartitionPath(this.hadoopConf, new Path(baseFile.getPath()));
-        } catch (Exception e) {
-          throw new HoodieException(String.format("Error when loading record keys from file: %s", baseFile), e);
-        }
 
-        for (HoodieKey hoodieKey : hoodieKeys) {
-          out.collect((O) new IndexRecord(generateHoodieRecord(hoodieKey, baseFile)));
-        }
+    Option<HoodieInstant> latestCommitTime = this.hoodieTable.getMetaClient().getCommitsTimeline()
+        .filterCompletedInstants().lastInstant();
+
+    if (latestCommitTime.isPresent()) {
+      List<FileSlice> fileSlices = this.hoodieTable.getSliceView()
+          .getLatestFileSlicesBeforeOrOn(partitionPath, latestCommitTime.get().getTimestamp(), true)
+          .collect(toList());
+
+      for (FileSlice fileSlice : fileSlices) {
+        // load parquet records
+        fileSlice.getBaseFile().ifPresent(baseFile -> {
+          // filter out crushed files
+          if (baseFile.getFileSize() <= 0) {
+            return;
+          }
+
+          if (shouldLoadFileId(baseFile.getFileId(), maxParallelism, parallelism, taskID)) {
+            LOG.info("Load records from file {}.", baseFile);
+            final List<HoodieKey> hoodieKeys;
+            try {
+              hoodieKeys =
+                  fileUtils.fetchRecordKeyPartitionPath(this.hadoopConf, new Path(baseFile.getPath()));
+            } catch (Exception e) {
+              throw new HoodieException(String.format("Error when loading record keys from file: %s", baseFile), e);
+            }
+
+            for (HoodieKey hoodieKey : hoodieKeys) {
+              out.collect((O) new IndexRecord(generateHoodieRecord(hoodieKey, baseFile)));
+            }
+          }
+        });
+
+        // load avro log records
+        fileSlice.getLogFiles().forEach(logFile -> {
+          // filter out crushed files
```

Comment:
You can read the logs all together with the scanner.

Reply:
ok, I already modify this.